find_overlay_dups

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Prints packages which are in both portage-stable and coreos-overlay
+
+SCRIPT_ROOT=$(dirname $(readlink -f "$0"))
+. "${SCRIPT_ROOT}/common.sh" || exit 1
+
+DEFINE_string overlay_path "${SRC_ROOT}/third_party/coreos-overlay" \
+    "Directory containing the overlay"
+DEFINE_string portage_stable_path "${SRC_ROOT}/third_party/portage-stable" \
+    "Path to portage-stable"
+
+# Parse flags
+FLAGS "$@" || exit 1
+eval set -- "${FLAGS_ARGV}"
+
+function get_tree_packages() {
+	# gets a list of all packages in a tree
+	find "$1" -type f -name "*.ebuild" -printf "%P\n" | xargs dirname | uniq
+}
+
+portage_stable_packages=$(get_tree_packages ${FLAGS_portage_stable_path})
+overlay_packages=$(get_tree_packages ${FLAGS_overlay_path})
+
+all_packages="$portage_stable_packages $overlay_packages"
+dups=$(sort <<< "$all_packages" | uniq -D | uniq)
```

Comment:
If you want to avoid storing long lists and writing a tmpfile, I think this does the same thing:

    find -L "${FLAGS_portage_stable_path}" "${FLAGS_overlay_path}" -maxdepth 3 -type f -name '*.ebuild' -printf '%P\n' | sed s,/[^/]*$,, | uniq | sort | uniq -d

Reply:
Does find always do a DFS or is that an implemenation detail. I don't see anything in `man find` defining a traversal order.

Reply:
The man page says the `depth` option uses depth-first ordering, although I don't think that affects this case.  All the ebuilds would be at the same depth.

Reply:
:+1: Although if a package is last in `portage-stable` and first in `coreos-overlay`, it would be skipped.